sw/device/tests/pwrmgr_sleep_disabled_test.c

```diff
@@ -0,0 +1,154 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+
+static volatile dif_aon_timer_irq_t irq;
+static volatile top_earlgrey_plic_peripheral_t peripheral;
+
+bool interrupt_serviced;
+bool interrupt_failed;
+
+bool is_pwrmgr_irq_pending(void) {
+  bool status;
+  CHECK_DIF_OK(dif_rv_plic_irq_is_pending(
+      &plic, kTopEarlgreyPlicIrqIdPwrmgrAonWakeup, &status));
+  return status;
+}
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  dif_rv_plic_irq_id_t irq_id;
+  if (is_pwrmgr_irq_pending()) {
```

Comment:
just curious, do you need to check for `pwrmgr` irq_id specifically? because looks like below it shows that any time it's NOT `aon_timer` an error is registered.

Reply:
It is not clear to me how we deal with multiple pending irq_id's, and this way seems to cut through confusion.